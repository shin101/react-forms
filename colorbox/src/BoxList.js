import React, { useState } from "react";


// Place your state that contains all of the boxes here. This component should render all of the Box components along with the NewBoxForm component

// const BoxList = () => {
//   const [color, setColor] = useState("grey");
//   return (
//     <form>
//       <Box />
//     </form>
//   )
// }

const NewBoxForm = () => {
  const initialState = {
    width: 100,
    height: 100,
    bgColor: 'white'
  }
  const [formData, setFormData] = useState(initialState)
  const handleChange = e => {
    const {name, value} = e.target;
    setFormData(data => ({
      ...data, 
      [name]: value
    }))
  }
  

  const handleSubmit = (e) => {
    e.preventDefault();
    const {width, height, bgColor} = formData;
    setFormData(initialState);
  }


  return (
    <form onSubmit = {handleSubmit}>
      <label htmlFor="width"></label>
      <input 
      id="width" 
      placeholder="width" 
      name="width"
      value={formData.width} 
      onChange={handleChange}/>

      <label htmlFor="height"></label>
      <input 
      id="height" 
      placeholder="height" 
      name="height"
      value={formData.height} 
      onChange={handleChange}/>
      
      <label htmlFor="bgColor"></label>
      <input 
      id="bgColor" 
      placeholder="bgColor" 
      name="bgColor"
      value={formData.bgColor} 
      onChange={handleChange}/>
      

      <button>Add a colorbox</button>
    </form>
  )
}




export default NewBoxForm;