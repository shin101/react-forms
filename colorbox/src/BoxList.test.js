import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import BoxList from './BoxList';


// smoke test
it("renders without crashing", function() {
  render(<BoxList />);
});

// snapshot
it("matches snapshot", function(){
  const {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
})

it("should not allow a user to delete anything when there is no box", function(){
  const {queryByText} = render(<BoxList />);

  expect(queryByText("X")).not.toBeInTheDocument();

});