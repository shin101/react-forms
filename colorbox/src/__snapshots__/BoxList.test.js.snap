// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Color Box Maker
    </h2>
    <form>
      <label
        for="width"
      />
      <input
        id="width"
        name="width"
        placeholder="width"
        value="100"
      />
      <label
        for="height"
      />
      <input
        id="height"
        name="height"
        placeholder="height"
        value="100"
      />
      <label
        for="bgColor"
      />
      <input
        id="bgColor"
        name="bgColor"
        placeholder="bgColor"
        value="black"
      />
      <button>
        Add a colorbox
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
